resolver/resolver.cc

```diff
@@ -3299,8 +3299,33 @@ class ResolveSignaturesWalk {
         send->addPosArg(ast::MK::Symbol(send->loc, method.data(ctx)->name));
     }
 
-    static void fillInInfoFromSig(core::MutableContext ctx, core::MethodRef method, core::LocOffsets exprLoc,
-                                  ParsedSig &sig, bool isOverloaded, const ast::MethodDef &mdef) {
+    struct ArgInformation {
+        std::vector<core::ArgInfo> posArgs;
+        std::vector<core::ArgInfo> kwArgs;
+        std::optional<core::ArgInfo> blkArg;
+    };
+
+    // This structure serves double duty: it holds information about a single sig and the
+    // results of running it through fillInInfoFromSig and it also holds information about
+    // merging the results of several such structures together from overloaded methods.
+    struct SigInformation {
```

Comment:
I worry about having something that is as generically named as `SigInformation` as we already have `ParsedSig` and also something else called `ArgInfo`. It seems like this structure is meant to hold information specifically for dealing with overloads + block args and I wonder whether we could make that more apparent in the name?